test/e2e/cws-tests/tests/test_e2e_cws_kubernetes.py

```diff
@@ -88,15 +75,43 @@ def test_open_signal(self):
         with Step(msg="check system-probe start", emoji=":customs:"):
             wait_agent_log("system-probe", self.kubernetes_helper, SYS_PROBE_START_LOG)
 
+        with Step(msg="check ruleset_loaded", emoji=":delivery_truck:"):
+            event = self.App.wait_app_log("rule_id:ruleset_loaded")
+            attributes = event["data"][-1]["attributes"]["attributes"]
+            start_date = attributes["date"]
+            self.App.check_for_ignored_policies(attributes)
+
         with Step(msg="wait for host tags (3m)", emoji=":alarm_clock:"):
             time.sleep(3 * 60)
 
-        with Step(msg="upload policies", emoji=":down_arrow:"):
-            self.kubernetes_helper.cp_to_agent("system-probe", self.policies, "/tmp/runtime-security.d/default.policy")
+        with Step(msg="check policies download", emoji=":file_folder:"):
+            self.policies = self.kubernetes_helper.download_policies()
+            self.assertNotEqual(self.policies, "", msg="check policy download failed")
+            data = self.policy_loader.load(self.policies)
+            self.assertIsNotNone(data, msg="unable to load policy")
+
+        with Step(msg="check rule presence in policies", emoji=":bullseye:"):
+            rule = self.policy_loader.get_rule_by_desc(desc)
+            self.assertIsNotNone(rule, msg="unable to find e2e rule")
+            self.assertEqual(rule["id"], agent_rule_name)
 
-        with Step(msg="restart system-probe", emoji=":rocket:"):
-            self.kubernetes_helper.kill_agent("system-probe", "-HUP")
-            time.sleep(60)
+        with Step(msg="push policies", emoji=":envelope:"):
+            self.kubernetes_helper.push_policies(self.policies)
+
+        with Step(msg="reload policies", emoji=":rocket:"):
+            self.kubernetes_helper.reload_policies()
+
+        with Step(msg="check ruleset_loaded", emoji=":delivery_truck:"):
+            for i in range(1, 60):  # retry 60 times
+                self.assertNotEqual(i, 59)  # timeout
```

Comment:
Not a huge fan of this, can this be replaced with https://book.pythontips.com/en/latest/for_-_else.html#else-clause or a simpler control flow ?

Reply:
Applied in last commit. Thanks.